src/main/java/fi/vm/yti/terminology/api/validation/VocabularyNodeValidator.java

```diff
@@ -0,0 +1,78 @@
+package fi.vm.yti.terminology.api.validation;
+
+import fi.vm.yti.terminology.api.model.termed.GenericNode;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.lang.annotation.*;
+
+public class VocabularyNodeValidator implements
+        ConstraintValidator<ValidVocabularyNode, GenericNode>, Annotation {
+
+    private boolean constraintViolationAdded;
+
+    @Override
+    public Class<? extends Annotation> annotationType() {
+        return null;
+    }
+
+    @Override
+    public boolean isValid(GenericNode genericNode, ConstraintValidatorContext context) {
```

Comment:
Validation for status and terminologyType would be good to add. Those are mandatory fields and they have predefined values (see Status and TerminologyType enums)